tests/integration/components/bs-form/element-test.js

```diff
@@ -1450,13 +1451,11 @@ module('Integration | Component | bs-form/element', function (hooks) {
   );
 
   test('it uses custom control component when registered in DI container', async function (assert) {
-    this.owner.register(
-      'component:bs-form/element/control/foo',
-      class extends Component {
-        tagName = '';
-        layout = hbs`<div id='foo'></div>`;
-      },
-    );
+    // eslint-disable-next-line ember/no-empty-glimmer-component-classes
+    class TestComponent extends Component {}
+    setComponentTemplate(hbs`<div id='foo' />`, TestComponent);
```

Comment:
Did you tried using `templateOnly` instead of an empty class? It should work accordingly to [RFC 481](https://github.com/emberjs/rfcs/blob/master/text/0481-component-templates-co-location.md#template-only-components) which introduced `setComponentTemplate`.